src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs

```diff
@@ -181,86 +169,69 @@ public virtual void ApplyResult(CorsResult result, HttpResponse response)
                 throw new ArgumentNullException(nameof(response));
             }
 
-            var headers = response.Headers;
-
-            if (result.AllowedOrigin != null)
-            {
-                headers[CorsConstants.AccessControlAllowOrigin] = result.AllowedOrigin;
-            }
-
-            if (result.VaryByOrigin)
+            if (!result.IsCorsResponseAllowed)
             {
-                headers["Vary"] = "Origin";
+                // In case a server does not wish to participate in the CORS protocol, its HTTP response to the 
+                // CORS or CORS-preflight request must not include any of the above headers. 
```

Comment:
Where are we logging that the preflight was denied?

Reply:
Happens as part of `IsOriginAllowed` which says the policy failed